### ISSUE TITLE
Macosx brew failed

### ISSUE THREAD

**Issue by N00D13** [redox-os/redox#40]
```
checking for gnatbind... no
checking for gnatmake... no
checking whether compiler driver understands Ada... no
checking how to compare bootstrapped objects... cmp --ignore-initial=16 $$f1 $$f2
checking for objdir... .libs
checking for the correct version of gmp.h... no
configure: error: Building GCC requires GMP 4.2+, MPFR 2.4.0+ and MPC 0.8.0+.
Try the --with-gmp, --with-mpfr and/or --with-mpc options to specify
their locations.  Source code for these libraries can be found at
their respective hosting sites as well as at
ftp://gcc.gnu.org/pub/gcc/infrastructure/.  See also
http://gcc.gnu.org/install/prerequisites.html for additional info.  If
you obtained GMP, MPFR and/or MPC from a vendor distribution package,
make sure that you have installed both the libraries and the header
files.  They may be located in separate packages.

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/homebrew):
  https://github.com/altkatz/homebrew-gcc_cross_compilers/issues
```

Anyone have a solution?

**Comment by jackpot51** [comment]
Which command did you run. Did you try to setup libc?

I don't think you can build the C cross compiler on homebrew at the moment, but it is not a critical piece of the system.

**Comment by ambaxter** [comment]
I was able to successfully compile everything from scratch. I had to add some entries in the osx-homebrew.sh. It should look like:

#!/bin/bash
brew tap homebrew/versions
brew install gcc49
brew tap altkatz/homebrew-gcc_cross_compilers
brew install i386-elf-binutils nasm

**Comment by jackpot51** [comment]
Can you verify that it is correct now?

**Comment by Nashenas88** [comment]
I'm getting this issue now (first time trying to setup redox). My output looks exactly the same as @N00D13's original comment. I got this error after running bootstrap.sh, and I also tried installing the required items manually to no luck. What output could I add here to help you narrow it down?

**Comment by Nashenas88** [comment]
I found a way around the issue by forking altkatz/homebrew-gcc_cross_compilers and modifying i386-elf-gcc.rb to point to gcc 4.9.3 (altkatz's version points to 4.9.2 while it's indirect depency on gcc49 builds 4.9.3) and also copying the following lines:
      "--with-gmp=#{Formula["gmp"].opt_prefix}",
      "--with-mpfr=#{Formula["mpfr"].opt_prefix}",
      "--with-mpc=#{Formula["libmpc"].opt_prefix}"
into the config args of i386-elf-gcc.rb as well.

I'm not sure how this will help with future proofing the project, but you could have bootstrap point to my version instead of altkatz's for the time being if you're comfortable with that. https://github.com/Nashenas88/homebrew-gcc_cross_compilers

**Comment by DomThePorcupine** [comment]
@Nashenas88 after using your new version does redox build?

**Comment by jackpot51** [comment]
Reopening, please can someone with OSX update tge scripts in a PR?

**Comment by Nashenas88** [comment]
@DomThePorcupine Yes, it does. I fired it up with `make qemu kvm=no` and it worked (very slowly, but that's on my end).

**Comment by Nashenas88** [comment]
Looks like this was fixed with PR #555 though it seems there was duplicated code there and the osx-homebrew.sh file isn't being called anymore. Should it be called or should the file be removed?

**Comment by DomThePorcupine** [comment]
I left it in incase people dont use the bootstrap and want to install everything manually. I guess looking back it doesnt make a whole lot of sense to have changed both but i guess it isnt really hurting anyone.
